docs/howtos/extending/magicgui.md

```diff
@@ -526,15 +615,46 @@ widget2 = my_factory(call_button=False, x={'widget_type': 'Slider'})
 
 :::
 
+
+(widget-classes)=
+
+## Widget classes
+
+Generating a widget by creating a widget class allows you to have more control over
+your widget. Your widget class must subclass {class}`magicgui.widgets.Widget` (i.e.,
+a [`magicgui` widget class](https://pyapp-kit.github.io/magicgui/widgets/#the-widget-hierarchy))
+or {class}`QtWidgets.QWidget`. It can then be added to the `napari` viewer
+by instantiating the widget class, then adding this via
+{meth}`napari.qt.Window.add_dock_widget`. You can also create a plugin and add
+your widget class (*not* instantiated widget) as a
+[widget contribution](widgets-contribution-guide).
+
+Below we will detail how to use various parent classes to generate a widget.
+There are several `magicgui` widget classes so we will only document the use of the
+two most useful in the `napari` context; {class}`magicgui.widgets.FunctionGui` and
+{class}`magicgui.widgets.Container`.
+We will begin with the simplest but least extensible parent class and end with the
+parent class the most extensible.
+
 ### `magicgui.widgets.FunctionGui`
 
-The other option for using `magicgui` in plugins is to directly subclass
-{class}`magicgui.widgets.FunctionGui` (which is the type that is returned
-by the {func}`@magicgui <magicgui.magicgui>` decorator).
+Creating a widget by subclassing {class}`magicgui.widgets.FunctionGui` is similar in
+principal to using the {func}`@magicgui <magicgui.magicgui>` decorator. Decorating
+a function with {func}`@magicgui <magicgui.magicgui>` is equivalent to passing
+the same function to {class}`magicgui.widgets.FunctionGui`'s `function` parameter.
+The remaining {class}`magicgui.widgets.FunctionGui` parameters essentially
+mirror {func}`@magicgui <magicgui.magicgui>`'s parameters.
+Indeed, {class}`magicgui.widgets.FunctionGui` is the type that is returned by
+{func}`@magicgui <magicgui.magicgui>`. Subclassing
+{class}`magicgui.widgets.FunctionGui` however, gives you access to the
+`native` `QWidget` of your widget, allowing you change its appearance and add
```

Comment:
Can `FunctionGui` guis receive the viewer as a parameter?

Reply:
Checked with the plugins PR (#4991) and yes it can. Does that affect it accessing the native `QWidget` ?!

Reply:
nope, doesn't affect that, I was just curious. I think a big benefit of the "class based" vs. "function based" widgets is that you can access the viewer and connect more complex events/callbacks in a simpler and more direct way. Function based widgets can also receive the viewer and do all that stuff, but the code ends up being more spaghettified

Reply:
I think magicgui functions are able to access the viewer! Its done via `magicgui.register_type`

Reply:
They are indeed. I just find when looking at plugin code that there's a correlation between function based widgets that take the viewer and needlessly complex logical flow - I suspect in many of these cases using a class based design would simplify things. But that's kinda hard to capture in this one doc lol so you can mostly ignore this comment chain for the purposes of review

Reply:
>  plugin code that there's a correlation between function based widgets that take the viewer and needlessly complex logical flow

I had no idea, why is it more complicated to use the viewer via a function?